File: lessons/lesson6.py


if __name__ == '__main__':
    # indexing a string
    s = '0123456'
    print(s[1])
    print(s[2])
    print(s[3])
    print('Begin!', '\n')

    # length of string
    i = len(s)
    print('length of', s, 'is', i)

    # substring
    s2 = s[1:i]
    print('s2 is', s2)

    # length of list of strings
    a = [str]
    a.append('0')
    a.append('1')
    a.append('2')
    a.append('3')
    a.append('4')
    i2 = len(a)

    # indexing list of strings
    a2 = a[3:i2]
    print(a2[0])

    # find in string
    s3 = '23'
    i3 = s.find(s3)
    print('position of 23:', i3)

    # find in list of strings
    s4 = '3'
    i4 = a.find(s4)
    print('position of 3:', i4)

File: lessons/lesson6.cpp
#include<vector>
#include<iostream>
#include<algorithm>
#include<string>


;
int main()
{
    // indexing a string
    std::string s("0123456");
    std::cout << s[1] << std::endl;
    std::cout << s[2] << std::endl;
    std::cout << s[3] << std::endl;
    std::cout << "Begin!"<< " " <<  "\n" << std::endl;
    // length of string
    auto i = (s).size();
    std::cout << "length of"<< " " <<  s<< " " <<  "is"<< " " <<  i << std::endl;
    // substring
    auto s2 = s.substr(1, i);
    std::cout << "s2 is"<< " " <<  s2 << std::endl;
    // length of list of strings
    auto a = std::vector<std::string>();
    a.push_back("0");
    a.push_back("1");
    a.push_back("2");
    a.push_back("3");
    a.push_back("4");
    auto i2 = (a).size();
    // indexing list of strings
    std::vector<std::string> a2(a.begin() + 3, a.begin() + i2);
    std::cout << a2[0] << std::endl;
    // find in string
    int i3 = s.find("23");
    std::cout << "position of 23:"<< " " <<  i3 << std::endl;
    // find in list of strings
    int i4 = std::find(a.begin(), a.end(), "3") - a.begin();
    std::cout << "position of 3:"<< " " <<  i4 << std::endl;
}